src/lib/utils/member.ts

```diff
@@ -39,117 +39,141 @@ export const getCustomAuthorOptions = async (
+              },
+              {
+                startDatetime: {
+                  lte: new Date(),
+                },
+              },
+            ],
+          },
+          {
+            // is active, or indefinite
+            OR: [
+              {
+                endDatetime: null,
+              },
+              {
+                endDatetime: {
+                  gte: new Date(),
+                },
+              },
+            ],
+          },
+          {
+            OR: [
+              {
+                studentId, // is for this user
+              },
+              {
+                role: {
+                  in: getDerivedRoles(
+                    memberPositionIds.map((pos) => pos.id),
+                    true,
+                  ).concat(
+                    memberPositionIds.some((pos) => pos.boardMember)
+                      ? ["dsek.styr"]
+                      : [],
+                  ),
+                },
+              },
+            ],
+          },
+        ],
       },
     })
-  ).concat(member.mandates.map((m) => m.position));
-
-  // Map a doorname to roles, startDate and endDate
-  const allMemberDoors = new Map<
-    string,
-    {
-      roles: string[];
-      startDate: Date | null;
-      endDate: Date | null;
-    }
-  >();
+    .catch(() => {
+      throw error(500, "Could not fetch door access");
+    });
 
-  allDoorPolicies
-    .filter((doorPolicy) =>
-      member.mandates.some(
-        (mandate) =>
-          // A doorpolicy is associated with either a role or a specific member
-          (doorPolicy.role && mandate.positionId.startsWith(doorPolicy.role)) ||
-          doorPolicy.studentId === member.studentId,
-      ),
-    )
-    .forEach((doorPolicy) => {
-      // Get a nice name for a position instead of using the id
-      const positionNamesFromMandates = positions
-        .filter(
-          (position) =>
-            doorPolicy.role && position.id.startsWith(doorPolicy.role),
+  const policiesByDoor: MemberDoorPolicies = userDoorPolicies.reduce(
+    (acc, policy) => {
+      const role = policy.role ?? "Du";
+      const duplicate = acc.find(
+        (p) =>
+          p.name === policy.doorName &&
+          p.startDate === policy.startDatetime &&
+          p.endDate === policy.endDatetime,
+      );
+      if (duplicate) {
+        duplicate.roles.push(role);
+        return acc;
+      }
+      acc.push({
+        name: policy.doorName,
+        roles: [role],
+        startDate: policy.startDatetime,
+        endDate: policy.endDatetime,
+      });
+      return acc;
+    },
+    [] as MemberDoorPolicies,
+  );
+  const memberDoorPolicies: MemberDoorPolicies = policiesByDoor.map(
+    (policy) => {
+      const positionsMappedToThisDoor = memberPositionIds
+        .filter((pos) =>
+          policy.roles.some(
+            (role) =>
+              pos.id.startsWith(role) ||
+              (pos.boardMember && role === "dsek.styr"),
```

Comment:
I don't really understand why "dsek.styr" is in the database in the first place. There's only one entry with that role, for "mauer", and it looks like all positions where "boardMember" is true already have access to mauer. Is there some other purpose for it or could it be removed?

Reply:
`dsek.styr` is used in more places I think, at least for api access it is used quite a bit. Maybe not as much for doors